aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,144 @@
+if (responseCachingFeature != null)
+{
+    responseCachingFeature.VaryByQueryKeys = new[] { "MyKey" };
+}
+```
+You can also create a custom `ResponseCachingFeature` that will apply `VaryByQueryKeys` globally to your application.
+```c#
+public void Configure(IApplicationBuilder app)
+{
+    app.UseResponseCaching();
+    
+    app.Use((context, next) =>
+    {
+        if (context.Features.Get<IResponseCachingFeature>() == null)
+        {
+            context.Features.Set<IResponseCachingFeature>(new CustomResponseCachingFeature());
+        }
+        return next();
+    });
+}
+
+public class CustomResponseCachingFeature : IResponseCachingFeature
+{
+    public string[] VaryByQueryKeys
+    {
+        get { return new[] { "MyKey" }; }
+        set { }
+    }
+}
+```
+
+## Caching conditions
+* The request must result in a 200 (OK) response from the server.
+* The request method must be GET or HEAD.
+* Terminal middlewares, such as Static File Middleware, must not process the response prior to the Response Caching Middleware.
+* The Authorization header must not be present.
+* `Cache-Control` header parameters must be valid and the response must be marked `public`.
+* The `Pragma: no-cache` header/value must not be present.
+* The `Set-Cookie` header must not be present.
+* `Vary` header parameters must be valid and not equal to `*`.
+* The `Content-Length` header (if set) must match the size of the response body.
+* The `HttpSendFileFeature` is not used.
+* Response buffering is successful.
+* The response must be cacheable according to the [RFC 7234](https://tools.ietf.org/html/rfc7234) specifications. For example, the `no-store` directive must must not exist in request or response header fields. See *Section 3: Storing Responses in Caches* and *Section 4.2: Freshness* of the RFC document for details.
+
+>[!NOTE]
+> The Antiforgery system for generating secure tokens to prevent Cross-Site Request Forgery (CSRF) attacks will set the `Cache-Control` and `Pragma` headers to `no-cache` so that responses will not be cached.
+
+## HTTP response caching headers
+Response caching by the middleware is configured via your HTTP response headers. The relevant headers are listed below with notes on how they affect caching.
+
+Header | Details
+--- | --- |
+Authorization | <p>The response is not cached if the header exists.</p>
+Cache-Control | <p>The header must be present and its value must explicitly be marked `public`.</p><p>You can control caching with the following parameters:</p><ul><li>max-age</li><li>max-stale</li><li>min-fresh</li><li>must-revalidate</li><li>no-cache</li><li>no-store</li><li>only-if-cached</li><li>private</li><li>public</li><li>s-maxage</li><li>proxy-revalidate</li></ul>
+Pragma | <p>The value must not be `no-cache`. This header is overridden by the relevant directives in the `Cache-Control` header if present.</p><p>Considered for backward compatibility with HTTP/1.0.</p>
```

Comment:
It can be `no-cache`. We just don't cache it if it's set. Let's use the same language as the other sections: "The response is not cached if the value is set to `no-cache`."